ethereum/contracts/Bridge.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./Decoder.sol";
+import "./Application.sol";
+
+contract Bridge {
+    using Decoder for bytes;
+
+    // 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470 is returned for accounts without code
+    bytes32 NON_CONTRACT_ACCOUNT_HASH = 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470;
+
+    mapping(bytes32 => bool) public eventHashes;
+    mapping(address => bool) public applications;
+
+    constructor(address[] memory apps) public {
+        for(uint256 i = 0; i < apps.length; i++) {
+            if(verifyApp(apps[i])) {
+                applications[apps[i]] = true;
+            }
+        }
+    }
+
+    /**
+     * @dev routes the message to the specified application ID after verifying the operator's signature
+     * @param _message address _message expected type: Message { AppID [32]byte, Payload []byte }
+     * @param _hash bytes _hash is a unique hash of the event (app_id, payload, blockNum, eventId)
+     */
+    function submit(bytes memory _message, bytes32 _hash)
```

Comment:
The signature should be `submit(address _appId, bytes memory _message)`, as this what the relayer is currently expecting.

Is there a mechanism in Solidity to hash the message?

Reply:
In Solidity we can hash with `keccak256(abi.encodePacked("one", "two", "three"));`

Reply:
hashed the message on contract in https://github.com/Snowfork/polkadot-ethereum/pull/75/commits/0f905430101dc05b1624c155753ac5169d4f6624